x-pack/plugins/cases/public/client/helpers/apm/use_cases_transactions.ts

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { Transaction } from '@elastic/apm-rum';
+import { useCallback } from 'react';
+import { CommentType } from '../../../../common/api/cases/comment';
+import { CaseAttachmentsWithoutOwner } from '../../../types';
+import { useStartTransaction } from './use_start_transaction';
+
+const CREATE_CASE = 'createCase' as const;
+const ADD_ATTACHMENT_TO_NEW_CASE = 'addAttachmentToNewCase' as const;
+const BULK_ADD_ATTACHMENT_TO_NEW_CASE = 'bulkAddAttachmentsToNewCase' as const;
+const ADD_ATTACHMENT_TO_EXISTING_CASE = 'addAttachmentToExistingCase' as const;
+const BULK_ADD_ATTACHMENT_TO_EXISTING_CASE = 'bulkAddAttachmentsToExistingCase' as const;
+
+export type StartCreateCaseWithAttachmentsTransaction = (param: {
+  appId: string;
+  attachments?: CaseAttachmentsWithoutOwner;
+}) => Transaction | undefined;
+
+// Called when a case is created, attachments are optional
+export const useCreateCaseWithAttachmentsTransaction = () => {
+  const { startTransaction } = useStartTransaction();
+
+  const startCreateCaseWithAttachmentsTransaction =
+    useCallback<StartCreateCaseWithAttachmentsTransaction>(
+      ({ appId, attachments }) => {
+        if (!attachments) {
+          return startTransaction(`Cases [${appId}] ${CREATE_CASE}`);
+        }
+        const alertCount = getAlertCount(attachments);
+        if (alertCount <= 1) {
+          return startTransaction(`Cases [${appId}] ${ADD_ATTACHMENT_TO_NEW_CASE}`);
+        }
+
+        const transaction = startTransaction(`Cases [${appId}] ${BULK_ADD_ATTACHMENT_TO_NEW_CASE}`);
+        transaction?.addLabels({ alert_count: alertCount });
+        return transaction;
```

Comment:
Setting the `alert_count` label to allow filtering in APM, ignore other attachments since alerts are the only type that can be "bulk" added.